### PR TITLE
Secondtable

### PR DESCRIPTION
**Pull request by rotshidzw**
- Use database transactions.
- Modify and delete data in SQL.
- Prepare complex queries that answer analytical questions.

avg
count id
data
delete
digimon
dont have species
first pic of updating the table
last of last
never escape
pokemon 2
pokemon
rollback delete
rollback
second commit
species mi  max
update negetive
verify
avg neatured
species change

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,9 +1,99 @@
...
+
+ROLLBACK;
+
+-- Delete all animals born after Jan 1st, 2022 & update weight to be non-negative
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT delete_after_2022;
+
+UPDATE animals SET weight_kg = weight_kg * -1;
+
```
Comment (Ghiftee) [inline data.sql]: - [ ] These transactions help analyze the data therefore they should not be in the ```data.sql``` file so kindly move all the transactions to the ```queries.sql``` file.

File: data.sql [hunk 2]
```diff
@@ -1,9 +1,99 @@
...
+         '2022-05-14',
+          22,
+           TRUE,
+            4
+    );
+
+
+BEGIN;
+
+UPDATE animals SET species = 'unspecified';
```
Comment (Ghiftee) [inline data.sql]: - [ ] After this change, according to the requirements, kindly ensure that the change was made before rolling it back. eg ```SELECT species from animals;```

File: data.sql [hunk 3]
```diff
@@ -1,9 +1,99 @@
...
+
+ROLLBACK;
+
```
Comment (Ghiftee) [inline data.sql]: - [ ] After rolling back the change, according to the requirements, kindly ensure that the change was indeed rolled back. eg ```SELECT species from animals;``` again

File: data.sql [hunk 4]
```diff
@@ -1,9 +1,99 @@
...
+
+-- Update the species column to digimon and pokemon depending on the name's suffix
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
```
Comment (Ghiftee) [inline data.sql]: - [ ] After this change, according to the requirements, kindly ensure that the change was made before committing it. eg ```SELECT species from animals;```

File: data.sql [hunk 5]
```diff
@@ -1,9 +1,99 @@
...
+COMMIT;
+
```
Comment (Ghiftee) [inline data.sql]: - [ ] After this commit, according to the requirements, kindly ensure that the commit persists. eg ```SELECT species from animals;```

File: data.sql [hunk 6]
```diff
@@ -1,9 +1,99 @@
...
+-- Delete all records and rollback
+BEGIN;
+
+DELETE FROM animals;
+
+ROLLBACK;
+
```
Comment (Ghiftee) [inline data.sql]: - [ ] After this change, according to the requirements, kindly verify if all records in the animals table still exists. eg ````SELECT COUNT(*) FROM ANIMALS;````
- [ ] Remember to ```BEGIN``` the new transaction below

**Review by Ghiftee — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️♻️
Hello 👋🏽  @rotshidzw 

Good job working on the project so far! 
✔️ Your queries work
✔️ There are no linter errors
However, there are some issues that you need to work on to go to the next project but you are almost there!



## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

### Hi Rotshidzwa, 
## Congratulations 🎉 Thank you for implementing all the requirements 🙏
### Please go ahead and merge the branch!

## Happy Coding 👏👏👏
#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**